components/experimental/src/dimension/provider/extended_currency.rs

```diff
@@ -0,0 +1,116 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+// Provider structs must be stable
+#![allow(clippy::exhaustive_structs, clippy::exhaustive_enums)]
+
+//! Data provider struct definitions for this ICU4X component.
+//!
+//! Read more about data providers: [`icu_provider`]
+
+use alloc::borrow::Cow;
+use icu_plurals::PluralCategory;
+use icu_provider::prelude::*;
+use tinystr::UnvalidatedTinyAsciiStr;
+use zerovec::{VarZeroVec, ZeroMap, ZeroMap2d, ZeroVec};
+
+#[cfg(feature = "compiled_data")]
+/// Baked data
+///
+/// <div class="stab unstable">
+/// 🚧 This code is considered unstable; it may change at any time, in breaking or non-breaking ways,
+/// including in SemVer minor releases. In particular, the `DataProvider` implementations are only
+/// guaranteed to match with this version's `*_unstable` providers. Use with caution.
+/// </div>
+pub use crate::provider::Baked;
+
+use super::currency::{CurrencyPatternConfig, PatternSelection, PlaceholderValue};
+
+/// Currency Extended V1 data struct.
+#[icu_provider::data_struct(marker(CurrencyExtendedDataV1Marker, "currency/extended@1"))]
+#[derive(Debug, Clone, Default, PartialEq)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize))]
+#[cfg_attr(
+    feature = "datagen", 
+    derive(serde::Serialize, databake::Bake),
+    databake(path = icu_experimental::dimension::provider::extended_currency)
+)]
+#[yoke(prove_covariance_manually)]
+pub struct CurrencyExtendedDataV1<'data> {
+    /// A mapping from each currency's ISO code to its associated formatting patterns.
+    // Using CurrencyPatternConfig until implementing AsULE for ExtendedCurrencyPatternConfig.
+    // use short for long right now.
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    pub patterns_config: ZeroMap<'data, Count, CurrencyPatternConfig>,
```

Comment:
Suggestion: I think in general the best way to store things that vary by plural form is to pull out the `other` variant and store all the others in the map. Something more like

```rust
/// For plural form 'other'
pub pattern_config_other: CurrencyPatternConfig,
/// For all other plural forms and explicit forms
pub pattern_config_plurals: ZeroMap<'data, Count, CurrencyPatternConfig>,
```

Reply:
great idea, we treated it as default pattern.